spec/GraphQLOverHTTP.md

```diff
@@ -112,13 +112,29 @@ http://example.com/product/graphql
 
 # Serialization Format
 
-Similar to the GraphQL specification this specification does not require a specific serialization format.
+The GraphQL specification allows for many [serialization formats to be implemented](https://spec.graphql.org/June2018/#sec-Serialization-Format). Servers and clients over HTTP MUST support JSON and MAY support other, additional serialization formats.
+
 For consistency and ease of notation, examples of the response are given in JSON throughout the spec.
 
+## Content Types
+
+The following are the officially recognized GraphQL content types to designate encoding JSON over HTTP.
+
+| Name | Description |
+| `application/graphql+json` | Required
+| `application/json` | To support legacy clients
+
+A server MUST support requests from clients with HTTP header `Content-Type: application/graphql+json` indicating that the body of the request is a JSON document with a GraphQL request. A server must indicate the content type of the response with a `Content-Type` header. This default value should be `Content-Type: application/graphql+json` indicating that the response is a valid JSON document containing at least one of the `data` or `errors` as a top-level attribute.
+
+A server MAY support requests from clients with other content types.
+
+A client MUST handle receiving responses with HTTP header `Content-Type: application/graphql+json` since any compliant server must support this type.
+
+This header MAY include encoding information (e.g. `Content-Type: application/graphql+json; charset=utf-8`)
+
 # Request
 
-The GraphQL HTTP server SHOULD handle the HTTP GET and POST methods.
-Additionally, GraphQL MAY be used in combination with other HTTP request methods.
+A server MUST accept POST requests, and MAY accept other HTTP methods, such as GET.
```

Comment:
A read-only GraphQL API that only uses persisted queries may use only GET; however by virtue of only supporting persisted queries its not really a full GraphQL server so I think we can keep this proposed change 👍 (otherwise POST is required as GET has size limits that vary wildly)

Reply:
The standard Introspection query would fit within a commonly used limit of 2048 characters, so that fundamental aspect of GraphQL could work. Still, for interoperability I think supporting POST is important. Otherwise it would complicate the implementation of clients, forcing them to have a fallback mechanism.